Use connections locally (#3225)
astronomer/docs: astro/cli/work-locally.md

Reviewer: Should we rename this file to `local-airflow-overview`?

---

> sidebar_label: 'Work locally'

Reviewer: ```suggestion
sidebar_label: 'Overview'
```

> title: 'Run Airflow locally'

Author: ```suggestion
title: 'Local Airflow overview'
```

> title: 'Run your Astro project locally'

Reviewer: ```suggestion
title: 'Run Airflow locally'
```

> id: work-locally

Reviewer: ```suggestion
id: local-airflow-overview
```

description: Work with your Astro project in a local environment by running Airflow and DAGs locally.
---

> Running Airflow locally with the Astro CLI can be an easy way to preview and debug DAG changes quickly before deploying your code to Astro. By locally running your DAGs, you can fix issues with your DAGs without consuming infrastructure resources or waiting on code deploy processes.

Reviewer: ```suggestion
Running Airflow locally with the Astro CLI lets you preview and debug DAG changes before deploying to production. In a local Airflow environment, you can fix issues with your DAGs without consuming infrastructure resources or waiting on code deploy processes. See the following documentation to get started:
```

> Running Airflow locally with the Astro CLI lets you preview and debug DAG changes before deploying to production. In a local Airflow environment, you can fix issues with your DAGs without consuming infrastructure resources or waiting on code deploy processes. See the following documentation to get started:

Reviewer: ```suggestion
Running Airflow locally with the Astro CLI lets you preview and debug DAG changes before deploying to production. In a local Airflow environment, you can fix issues with your DAGs without consuming infrastructure resources or waiting on code deploy processes. 

To run Airflow locally, the Astro CLI creates and runs containers for core Airflow components. It uses Docker by default to orchestrate these containers, but you can also use [Podman](cli/configure-cli.md#run-the-astro-cli-using-podman). All tasks run locally in the scheduler container using the [local executor](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/local.html).

See the following documentation to get started:
```
This seemed like a good opportunity to bubble up some information that's scattered around the docs about how exactly local environments work

- [Run Airflow locally](cli/run-airflow-locally.md)
- [Test your DAGs](cli/test-your-astro-project-locally.md)
- [Trouleshoot locally](cli/troubleshoot-locally.md)

> ## Sync connection details

Reviewer: ```suggestion
## Astro integrations
```
Or something similar? This would let us add more content to the section about deploying projects/ other Astro Environment Manager features without creating new topics

Author: We might want to get the product team's feedback on what to call this. Fritz had some feedback about it and I'm not confident I know the right terms to communicate the nuance

Reviewer: Maybe "Work with Astro-hosted objects"? That should cover most circumstances, and I think "object" is the most technically accurate term for what's actually being pulled down

Reviewer: I think it works for now, but I'd like to revisit it after the Environment Manager encapsulates more than just connections
